src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -52,33 +69,95 @@ export async function initialize({
             return await onLocalInvokeCommand({
                 commandName: command,
                 lambdaLocalInvokeParams: params,
+                configuration,
+                toolkitOutputChannel,
+                processInvoker,
+                taskInvoker,
+                telemetryService
             })
         },
     })
 }
 
 /**
  * The command that is run when user clicks on Run Local or Debug Local CodeLens
+ * Accepts object containing the following params:
+ * @param configuration - SettingsConfiguration (for invokeLambdaFunction)
+ * @param toolkitOutputChannel - "AWS Toolkit" output channel
  * @param commandName - Name of the VS Code Command currently running
  * @param lambdaLocalInvokeParams - Information about the Lambda Handler to invoke locally
+ * @param processInvoker - SAM CLI Process invoker
+ * @param taskInvoker - SAM CLI Task invoker
+ * @param telemetryService - Telemetry service for metrics
  */
 async function onLocalInvokeCommand({
+    configuration,
+    toolkitOutputChannel,
     commandName,
     lambdaLocalInvokeParams,
+    processInvoker,
+    taskInvoker,
+    telemetryService
 }: {
+    configuration: SettingsConfiguration
+    toolkitOutputChannel: vscode.OutputChannel,
     commandName: string,
     lambdaLocalInvokeParams: LambdaLocalInvokeParams,
+    processInvoker: SamCliProcessInvoker,
+    taskInvoker: SamCliTaskInvoker,
+    telemetryService: TelemetryService
 }): Promise<{ datum: Datum }> {
 
+    const channelLogger = getChannelLogger(toolkitOutputChannel)
     const runtime = await getRuntimeForLambda({
         handlerName: lambdaLocalInvokeParams.handlerName,
         templatePath: lambdaLocalInvokeParams.samTemplate.fsPath
     })
 
-    // TODO : Implement local run/debug in future backlog tasks
-    vscode.window.showInformationMessage(
-        `Local ${lambdaLocalInvokeParams.isDebug ? 'debug' : 'run'} support for csharp is currently not implemented.`
-    )
+    if (!lambdaLocalInvokeParams.isDebug) {
+        const baseBuildDir = await makeBuildDir()
+        const codeDir = path.dirname(lambdaLocalInvokeParams.document.uri.fsPath)
```

Comment:
What happens if the document is in a subfolder to the .csproj file?

Reply:
Started up a hello world template and moved everything but the `.csproj` file into a subfolder (`./nextLevel`). Output from a run:
```
Building SAM Application...
An error occurred trying to run SAM Application locally: sam build encountered an error: 2019-04-18 10:57:20 Building resource 'awsToolkitSamLocalResource'
,2019-04-18 10:57:21 Running DotnetCliPackageBuilder:GlobalToolInstall
,2019-04-18 10:57:21 
,2019-04-18 10:57:26 Tool 'amazon.lambda.tools' was reinstalled with the latest stable version (version '3.2.0').
,2019-04-18 10:57:26 Running DotnetCliPackageBuilder:RunPackageAction
,2019-04-18 10:57:26 Amazon Lambda Tools for .NET Core applications (3.2.0)

Project Home: https://github.com/aws/aws-extensions-for-dotnet-cli, https://github.com/aws/aws-lambda-dotnet

	

No .NET project found in directory C:\Users\XXXXXXXXX\dotnet-test\sam-app\asdf\src\HelloWorld\nextLevel to build.
,Error: DotnetCliPackageBuilder:RunPackageAction - Dotnet CLI Failed:
```

Taking a look...

Reply:
✔️